isaak-willett/Tawa
# Starts of Eototo (TawaCLI)

## Changes:

- Builds the entire first version of interfacing to Tawa from exterior
  - I'm going to link a justification document here from internal pages

## Motivation (if not covered in GH Issue):

- Creates dual interface for operating in and out of the container very hermetically

## Type of Change:
- [ ] Bug fix (non-breaking change which fixes and issue)
- [ ] Breaking change (fix or feature that changes expected behavior)
- [ ] Requires documentation update (please link to update)

## Checklist:

- [ ] Add issue link if associated
- [ ] Include MR comments to help with hard to review sections
- [ ] Write code in self-documenting style
- [ ] On Merge Request pass CI checks (or relevant reason for expected failures)
  * No need to pass on CI checks until MR/Review mode is put up
- [ ] If feature, add tests, if bug fix, update tests
- [ ] If feature, do we need to add analytics

## How to Test:

- Install eototo using `pip install -e Eototo`
- Run `eototo lint`
- See ruff in action in the container through the docker links
  - notice you're running in the cuda12 container
  - You're actually running `taw-cli lint` through the docker container
  - You're automatically picking up the internal cli commands without any dependency on the same packages outside the container
- Install Tawa using `pip install -e Tawa['dev']`
- Run `taw-cli lint`
- Notice you're running outside the container

## Demo:

### External:
https://github.com/isaak-willett/Tawa/assets/58192697/10900e0c-a446-4c96-b31b-bd0fb0628502

### Internal:
https://github.com/isaak-willett/Tawa/assets/58192697/20041b15-1b98-40d4-81b3-c7187b375663

### Easy Command Dev:
https://github.com/isaak-willett/Tawa/assets/58192697/e6d11c43-8acb-4dcb-a1dd-8dda3e4ffb5a

https://github.com/isaak-willett/Tawa/assets/58192697/92c6227a-4cdf-4b34-a09c-0572275b59dc
